backend/src/services/okr.service.ts

```diff
@@ -0,0 +1,82 @@
+import { getCustomRepository } from 'typeorm';
+import { OKR } from '~/data/entities/okr';
+import Okrepository from '~/data/repositories/okr.repository';
+import UserRepository from '~/data/repositories/user.repository';
+import { badRequestError } from '~/common/errors';
+
+export const getAllOkr = async (userId: string): Promise<OKR[]> => {
+  const okrRepository = getCustomRepository(Okrepository);
+  const okrs = okrRepository.getAllByUserId(userId);
+  return okrs;
+};
+
+export const getOkrById = async (okrId: string): Promise<OKR> => {
+  const okrRepository = getCustomRepository(Okrepository);
+
+  const okr = await okrRepository.findOne({ id: okrId });
+
+  if (okr) {
+    const okr = okrRepository.getOneByUserId();
+    return okr;
+  }
+
+  throw badRequestError('Okr isn`t exist!!!');
+};
+
+export const createOkr = async ({
+  userId,
+  body,
+}: {
+  userId: string;
+  body: {
+    name: string;
+    endDate: string;
+    startDate: string;
+  };
+}): Promise<OKR> => {
+  const okrRepository = getCustomRepository(Okrepository);
+  const userRepository = getCustomRepository(UserRepository);
+
+  if (!body.name) throw badRequestError('Okr name is undefined!!!');
+  if (!body.startDate) throw badRequestError('Okr startDate is undefined!!!');
+  if (!body.endDate) badRequestError('Okr endDate is undefined!!!');
```

Comment:
Since you use `validateBody` middleware you do not need these validations anymore